var/spack/repos/builtin/packages/py-torchgeo/package.py

```diff
@@ -58,7 +58,7 @@ class PyTorchgeo(PythonPackage):
         depends_on('py-h5py', type='run')
         depends_on('py-laspy@2:', when='@0.2:', type='run')
         depends_on('open3d@0.11.2:+python', when='@0.2:', type='run')
-        depends_on('opencv+python3+imgcodecs+tiff+jpeg+png', type='run')
+        depends_on('opencv+python3+imgcodecs+tiff+jpeg+png+datasets', type='run')
```

Comment:
Tried testing this. The only dataset in TorchGeo that uses OpenCV is the LandCover.ai dataset. It uses this to read TIFF files and write JPEG/PNG files, all in pure-Python. However, this seems to fail:
```console
$ spack install py-torchgeo+datasets
$ spack load py-torchgeo+datasets
$ python
>>> from torchgeo.datasets import LandCoverAI
>>> ds = LandCoverAI(download=True, checksum=True)
Downloading https://landcover.ai/download/landcover.ai.v1.zip to data/landcover.ai.v1.zip
1538212864it [05:22, 4766024.88it/s]                                                                                                                                      
Extracting data/landcover.ai.v1.zip to data
[ WARN:0] global /var/folders/21/hwq39zyj4g36x6zjfyl5l8080000gn/T/Adam/spack-stage/spack-stage-opencv-4.5.4-urkmryldyvjy5qmhjepejt6utj7qmnmg/spack-src/modules/imgcodecs/src/grfmt_tiff.cpp (457) readData OpenCV TIFF: TIFFRGBAImageOK: Sorry, requested compression method is not configured
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/Adam/torchgeo/torchgeo/datasets/landcoverai.py", line 109, in __init__
    self._verify()
  File "/Users/Adam/torchgeo/torchgeo/datasets/landcoverai.py", line 200, in _verify
    self._extract()
  File "/Users/Adam/torchgeo/torchgeo/datasets/landcoverai.py", line 234, in _extract
    exec(split)
  File "<string>", line 27, in <module>
AttributeError: 'NoneType' object has no attribute 'shape'
```
Is there something else we're missing here? Also, what made you add `+datasets`?

Reply:
I guess I made an assumption about `datasets`. That message is coming from `libtiff`.

Reply:
@glennpj this is a bit off-topic, but do you know how to determine which `libtiff` variants are needed? All I can tell is that it seems to use JPEG:
```console
$ file M-33-20-D-d-3-3.tif
M-33-20-D-d-3-3.tif: TIFF image data, little-endian
$ gdalinfo M-33-20-D-d-3-3.tif
...
Image Structure Metadata:
  COMPRESSION=YCbCr JPEG
  INTERLEAVE=PIXEL
  SOURCE_COLOR_SPACE=YCbCr
```
I've tried both `+jpeg` and `+old-jpeg` but I still can't read this file.

Reply:
All of the variants for `libtiff` are set to `False`. That is surprising, but then opencv is the same way. For libtiff, compression support will need to be enabled. I did not realize libtiff was set that way so opencv will need to depend on variants being set when using tiff.

Reply:
At a minimum, `libdeflate` is needed. I think `lzma` and `zlib` are going to be very common as well.

Reply:
Using `libtiff+jpeg+libdeflate`
```
$ python
Python 3.9.9 (main, Jan 20 2022, 15:07:33)
[GCC 10.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from torchgeo.datasets import LandCoverAI
>>> ds = LandCoverAI(download=True, checksum=True)
Downloading https://landcover.ai/download/landcover.ai.v1.zip to data/landcover.ai.v1.zip
1538212864it [01:20, 19074956.52it/s]
Extracting data/landcover.ai.v1.zip to data
Processed M-33-20-D-c-4-2 1/41
Processed M-33-20-D-d-3-3 2/41
Processed M-33-32-B-b-4-4 3/41
Processed M-33-48-A-c-4-4 4/41
Processed M-33-7-A-d-2-3 5/41
Processed M-33-7-A-d-3-2 6/41
Processed M-34-32-B-a-4-3 7/41
Processed M-34-32-B-b-1-3 8/41
Processed M-34-5-D-d-4-2 9/41
Processed M-34-51-C-b-2-1 10/41
Processed M-34-51-C-d-4-1 11/41
Processed M-34-55-B-b-4-1 12/41
Processed M-34-56-A-b-1-4 13/41
Processed M-34-6-A-d-2-2 14/41
Processed M-34-65-D-a-4-4 15/41
Processed M-34-65-D-c-4-2 16/41
Processed M-34-65-D-d-4-1 17/41
Processed M-34-68-B-a-1-3 18/41
Processed M-34-77-B-c-2-3 19/41
Processed N-33-104-A-c-1-1 20/41
Processed N-33-119-C-c-3-3 21/41
Processed N-33-130-A-d-3-3 22/41
Processed N-33-130-A-d-4-4 23/41
Processed N-33-139-C-d-2-2 24/41
Processed N-33-139-C-d-2-4 25/41
Processed N-33-139-D-c-1-3 26/41
Processed N-33-60-D-c-4-2 27/41
Processed N-33-60-D-d-1-2 28/41
Processed N-33-96-D-d-1-1 29/41
Processed N-34-106-A-b-3-4 30/41
Processed N-34-106-A-c-1-3 31/41
Processed N-34-140-A-b-3-2 32/41
Processed N-34-140-A-b-4-2 33/41
Processed N-34-140-A-d-3-4 34/41
Processed N-34-140-A-d-4-2 35/41
Processed N-34-61-B-a-1-1 36/41
Processed N-34-66-C-c-4-3 37/41
Processed N-34-77-A-b-1-4 38/41
Processed N-34-94-A-b-2-4 39/41
Processed N-34-97-C-b-1-2 40/41
Processed N-34-97-D-c-2-4 41/41
>>>
```

Reply:
It looks like `+jpeg+zlib` is required. I'll add that in a separate PR. Anyway, you can remove `+datasets` from this PR, it isn't needed.

Reply:
Hmm, I got it working with `+jpeg+zlib~libdeflate`, not sure if `+libdeflate` is actually required.

Yeah, the variants for `libtiff` are pretty bad. I'm the one who disabled everything by default in both OpenCV and libtiff. Trying to minimize building time for packages that only need a small portion of the library. Once we have public binaries we can expand this.

Reply:
Right, zlib is needed for libdeflate, and clingo enables zlib support with libdeflate.